signatures.go

```diff
@@ -0,0 +1,29 @@
+package distribution
+
+import "github.com/docker/distribution/digest"
+
+// WIP(stevvooe): While the other interfaces are based on active
+// specifications or proposals, the signatures interface still lacks a
+// complete proposal. We may need to add or remove functionality, likely on
+// the signature type itself.
+
+// Signature is simply a blob that may contain a cryptographic blob used for
+// verification. The contents are dependent on the trust system. The main
+// functionality of a signature comes from the relationships expressed in the
+// SignatureService interface.
+type Signature interface {
+	Blob // io access to the signature data itself.
+}
+
+// SignatureService provides the ability to get and set signatures for a
+// particular object. One or more signatures can be associated with a blob.
+type SignatureService interface {
+	// Get returns the signatures associated with the provided digest.
+	Get(dgst digest.Digest) ([]Signature, error)
+
+	// Put associates one or more signatures with specified digest.
+	Put(dgst digest.Digest, signatures ...Signature) error
+
+	// Delete removes one or more signatures from the targeted digest.
+	Delete(dgst digest.Digest, signatures ...Signature) error
```

Comment:
The delete function is intended to be given the full signature blob then remove each individual signature by the digest of the signature? The signature objects should be able to use their descriptors just like any other blob.  I assume the idea behind making it a blob is to easily allow different signature media type (PGP, JWS). If that is the case, why not just pass in signatures as digests.

I think this is still clear that dgst is the target.
~~~go
Delete(dgst digest.Digest, signatures ...digest.Digest) error
~~~